tools/qc/fastq_utils/fastq_pre_barcodes.xml

```diff
@@ -0,0 +1,227 @@
+<tool id="fastq_pre_barcodes" name="FASTQ barcodes preprocessor" profile="18.01" version="0.16.3+galaxy0">
+    <description>Preprocesses the reads to move the barcodes (UMI, Cell, ...) to the respective readname, optionally discarding reads with bases in the barcode regions below a given threshold.</description>
+    <requirements>
+        <requirement type="package" version="0.25.1">fastq_utils</requirement>
+    </requirements>
+    <command detect_errors="exit_code"><![CDATA[
+    fastq_pre_barcodes --read1 '$read1' --outfile1 '$outfile1'
+
+    #if $read2:
+        --read2 '$read2'
+    #end if
+
+    #if $index1:
+        --index1 '$index1'
+    #end if
+
+    #if $index2:
+        --index2 '$index2'
+    #end if
+
+    #if $index3:
+        --index3 '$index3'
+    #end if
+
+    #if $phred_encoding:
+        --phred_encoding '$phred_encoding'
+    #end if
+
+    #if $min_qual:
+        --min_qual '$min_qual'
+    #end if
+
+    #if $outfile2:
+        --outfile2 '$outfile2'
+    #end if
+
+    #if $outfile3:
+        --outfile3 '$outfile3'
+    #end if
+
+    #if $interleaved:
+        --interleaved '$interleaved'
+    #end if
+
+    #if $umi_read:
+        --umi_read '$umi_read'
+    #end if
+
+    #if $umi_offset:
+        --umi_offset '$umi_offset'
+    #end if
+
+    #if $umi_size:
+        --umi_size '$umi_size'
+    #end if
+
+    #if $cell_read:
+        --cell_read '$cell_read'
+    #end if
+
+    #if $cell_offset:
+        --cell_offset '$cell_offset'
+    #end if
+
+    #if $cell_size:
+        --cell_size '$cell_size'
+    #end if
+
+    #if $sample_read: 
+        --sample_read '$sample_read'
+    #end if
+
+    #if $sample_offset:
+        --sample_offset '$sample_offset'
+    #end if
+
+    #if $sample_size:
+        --sample_size '$sample_size'
+    #end if
+
+    #if $read1_offset:
+        --read1_offset '$read1_offset'
+    #end if
+
+    #if $read1_size:
+        --read1_size '$read1_size'
+    #end if
+
+    #if $read2_offset:
+        --read2_offset '$read2_offset'
+    #end if
+
+    #if $read2_size: 
+        --read2_size '$read2_size'
+    #end if
+
+    #if $use_10x:
+        '$use_10x'
+    #end if
+
+    #if $sam:
+        '$sam'
+    #end if
+
+    #if $x:
+        '$x'
+    #end if
+
+    #if $brief: 
+        '$brief'
+    #elif $verbose:
+        '$verbose'
+    #end if
+    ]]></command>
+    <inputs>
+        <param name="verbose" label="Verbose" optional="true" value='false' argument="--verbose" type="boolean"  truevalue='--verbose' falsevalue='' checked="true" help="Increase level of messages printed to stderr"/>
+        <param name="brief" label="Brief" optional="true" value="true" argument="--brief" type="boolean"  truevalue='--brief' falsevalue='' checked="true" help="Decrease level of messages printed to stderr"/>
+        <param name="read1" label="Read1"  argument="--read1" type="data" format="fastqsanger" optional="false" help="fastq (optional gzipped) file name"/>
```

Comment:
Also, if the tool actually accepts fastq.gz, I would try adding fastqsanger.gz here instead of fastqsanger (sorry, I know it was my original suggestion). What happens here is that if the tool can accept .gz, then galaxy is decompressing this unnecesarily to pass it as fastqsanger instead of fastqsanger.gz. On The inputs I think that you can accept more than one format (so you could use both, comma separated within the field).

Reply:
Ah, originally I put "fastq,fastqsanger", incorrectly interpreting that one of them stood for the .gz version, and this gave me an error.  Would you know where `format` values are documented?  [This page about data types](https://galaxyproject.org/learn/datatypes/) does not list the actual `format` values that should be used in the xml.

Reply:
Done in e9f8adb3fd9d4ae73e3b8c5d32f6dd4768a7f214.